x-pack/plugins/security_solution/public/common/components/sourcerer/index.tsx

```diff
@@ -136,37 +166,72 @@ export const Sourcerer = React.memo<SourcererComponentProps>(({ scope: scopeId }
     [setPopoverIsOpenCb, loading]
   );
 
+  const dataViewSelectOptions = useMemo(
+    () =>
+      kibanaDataViews.map(({ title, id }) => ({
+        inputDisplay:
+          id === defaultDataView.id ? (
+            <span data-test-subj="security-option-super">
+              <EuiIcon type="logoSecurity" size="s" /> {i18n.SIEM_DATA_VIEW_LABEL}
+            </span>
+          ) : (
+            <span data-test-subj="dataView-option-super">
+              <EuiIcon type="logoKibana" size="s" /> {title}
+            </span>
+          ),
+        value: id,
+      })),
+    [defaultDataView.id, kibanaDataViews]
+  );
+
   const comboBox = useMemo(
     () => (
       <EuiComboBox
-        data-test-subj="indexPattern-switcher"
-        placeholder={i18n.PICK_INDEX_PATTERNS}
+        data-test-subj="sourcerer-combo-box"
         fullWidth
-        options={indexesPatternOptions}
-        selectedOptions={selectedOptions}
         onChange={onChangeCombo}
+        options={selectableOptions}
+        placeholder={i18n.PICK_INDEX_PATTERNS}
         renderOption={renderOption}
+        selectedOptions={selectedOptions}
       />
     ),
-    [indexesPatternOptions, onChangeCombo, renderOption, selectedOptions]
+    [selectableOptions, onChangeCombo, renderOption, selectedOptions]
   );
 
+  const superSelect = useMemo(
```

Comment:
I do not think that we need to have useMemo. and we can just use the `EuiSuperSelect` directly. React.memo will do its job.

Reply:
done!